test/tcpBackoff.js

```diff
@@ -23,7 +23,37 @@ test('tcp backoff', function testTcpBackoff (done) {
       const nextBackoffDelay = exponentialStrategy.next()
       // initial, 10, 100... next delay should be 1000
       expect(nextBackoffDelay).to.eq(1000)
-      tcpConnection.end(() => done())
+      tcpConnection.end(() => {
+        process.stdin.removeAllListeners()
```

Comment:
I have to explicitly remove all listeners on `process.stdin` otherwise we get:

```
(node:32722) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 end listeners added to [ReadStream]. Use emitter.setMaxListeners() to increase limit
    at _addListener (node:events:595:17)
    at ReadStream.addListener (node:events:617:10)
    at ReadStream.Readable.on (node:internal/streams/readable:876:35)
    at ReadStream.once (node:events:661:8)
    at ReadStream.Readable.pipe (node:internal/streams/readable:678:9)
    at buildCliSourceStream (/path/to/pino-socket/lib/TcpConnection.js:252:17)
    at factory (/path/to/pino-socket/lib/TcpConnection.js:57:7)
    at Context.<anonymous> (/path/to/pino-socket/test/tcpSocketError.js:31:25)
    at callFnAsync (/path/to/pino-socket/node_modules/mocha/lib/runnable.js:394:21)
    at Test.Runnable.run (/path/to/pino-socket/node_modules/mocha/lib/runnable.js:338:7)
    at Runner.runTest (/path/to/pino-socket/node_modules/mocha/lib/runner.js:666:10)
    at /path/to/pino-socket/node_modules/mocha/lib/runner.js:789:12
    at next (/path/to/pino-socket/node_modules/mocha/lib/runner.js:581:14)
    at /path/to/pino-socket/node_modules/mocha/lib/runner.js:591:7
    at next (/path/to/pino-socket/node_modules/mocha/lib/runner.js:474:14)
    at Immediate.<anonymous> (/path/to/pino-socket/node_modules/mocha/lib/runner.js:559:5)
    at processImmediate (node:internal/timers:466:21)
```

Not sure if there's a better way to fix this issue...